core/crossmodel/offerurl_test.go

```diff
@@ -25,20 +25,20 @@ var urlTests = []struct {
 	exact  string
 	url    *crossmodel.OfferURL
 }{{
```

Comment:
I think we need a test for old offer URLs that could have had a user name. Especially one that flexes the full syntax of what could have been an user name (e.g. bob@external).

Even if the result of that test is an error. We should check it.

Reply:
There's a code comment that says it's up to the target controller to handle the passed in offer URLs and perform any compatibility transformations. The parsing client side is just to ensure that the URL syntax is correct and to validate things like model name which are unambiguous.